public/comments.php

```diff
@@ -22,135 +28,121 @@
+        /** @var System $console */
+        $console = $game->console;
+        $pageTitle = renderGameTitle($game->Title . ' (' . $console->Name . ')');
         $commentsLabel = "Modifications";
         $navPath =
         [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
+            '_GamePrefix' => renderGameBreadcrumb($game->ID),
         ];
         break;
 
     case ArticleType::SetClaim:
         if ($permissions < Permissions::Admin) {
             abort(403);
         }
-        $gameData = getGameData($articleID);
-        if ($gameData === null) {
-            abort(404);
-        }
-        $pageTitle = renderGameTitle($gameData['Title'] . ' (' . $gameData['ConsoleName'] . ')');
+        /** @var Game $game */
+        $game = Game::findOrFail($articleID);
+        /** @var System $console */
+        $console = $game->console;
+        $pageTitle = renderGameTitle($game->Title . ' (' . $console->Name . ')');
         $commentsLabel = "Claim Comments";
-        $navPath =
-        [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
-            'Manage Claims' => '/manageclaims.php?g=' . $gameData['ID'],
+        $navPath = [
+            '_GamePrefix' => renderGameBreadcrumb($game->ID),
+            'Manage Claims' => '/manageclaims.php?g=' . $game->ID,
         ];
         break;
 
     case ArticleType::Achievement:
-        $pageTitle = getAchievementTitle($articleID, $gameTitle, $gameID);
-        if (empty($pageTitle)) {
-            abort(404);
-        }
-        $gameData = getGameData($gameID);
-        if ($gameData === null) {
-            abort(404);
-        }
-        $navPath =
-        [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
-            $pageTitle => '/achievement/' . $articleID,
+        /** @var Achievement $achievement */
+        $achievement = Achievement::findOrFail($articleID);
+        /** @var Game $game */
+        $game = Game::findOrFail($achievement->GameID);
+        /** @var System $console */
+        $console = $game->console;
+        $pageTitle = $achievement->Title;
+        $navPath = [
+            '_GamePrefix' => renderGameBreadcrumb($game->ID),
+            $pageTitle => '/achievement/' . $achievement->ID,
         ];
         break;
 
     case ArticleType::Leaderboard:
-        $pageTitle = getleaderboardTitle($articleID, $gameID);
-        if (empty($pageTitle)) {
-            abort(404);
-        }
-        $gameData = getGameData($gameID);
-        if ($gameData === null) {
-            abort(404);
-        }
-        $navPath =
-        [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
-            $pageTitle => '/leaderboard/' . $articleID,
+        /** @var Leaderboard $leaderboard */
+        $leaderboard = Leaderboard::findOrFail($articleID);
+        /** @var Game $game */
+        $game = Game::findOrFail($leaderboard->GameID);
+        $console = $game->console;
```

Comment:
$console is unannotated, also unused

Reply:
Removed the unused variable. However, technically, it doesn't have to be annotated for static analysis at this point because the access on the Game object knows that it's a System instance.